pkg/test/framework/components/istio/eastwest.go

```diff
@@ -48,7 +48,7 @@ var (
 )
 
 // deployEastWestGateway will create a separate gateway deployment for cross-cluster discovery or cross-network services.
-func (i *operatorComponent) deployEastWestGateway(cluster cluster.Cluster, revision string) error {
+func (i *operatorComponent) deployEastWestGateway(cluster cluster.Cluster, revision string, customSettings string) error {
```

Comment:
I think it would be better if we handle this through existing `genGatewayScript` itself?
Either we can always add the volume mounts for pilot certs through the shell `gen-eastwest-gateway.sh` and if we dont want to mount those volumes always, we can add another flag like `--mountPilotCerts` or something.
We dont need to add new customisation. This is adding complexity unnecesssary

Reply:
The advantage to doing to it in this way is that we can handle any kind of update to eastwest gateway settings, we are anyway doing same for ingress and egress.

Reply:
If we want to make it more customizable than the just pilot certs, cant we pass arbitrary values to exisitng shell script?

Reply:
The shell script can't do smart yaml merging which could make things tricky. A human user might use this script to generate the "base" for an easy West gateway (which is really a specialized ingress) and then modify it.

This is the simplest/most flexible thing we can do for automated tests imo.